service/routes.go

```diff
@@ -5,17 +5,39 @@ import (
 	"github.com/rancher/go-rancher/api"
 	"github.com/rancher/go-rancher/client"
 	//This should be v2, supporting schemas
+	"github.com/rancher/webhook-service/drivers"
 	"net/http"
 )
 
 var schemas *client.Schemas
 var router *mux.Router
 
+type payload struct {
+	client.Resource
+	URL string `json:"url"`
+}
+
 func NewRouter() *mux.Router {
-	schemas = &client.Schemas{}
+	schemas = driverSchemas()
 	router = mux.NewRouter().StrictSlash(true)
+
 	router.Methods("POST").Path("/v1-webhooks-generate").Handler(api.ApiHandler(schemas, http.HandlerFunc(ConstructPayload)))
```

Comment:
You didn't add the error handling we talked about. The example is here: https://github.com/rancher/longhorn/blob/master/agent/controller/rest/router.go#L13-L52

Your handlers (in handler.go) should return an error and be wrapped by the function that converts errors into http json responses.